### PR TITLE
UiTdatabank - Search API documentation

### PR DESCRIPTION
**Pull request by corneelwille**
### Added
- Sorting guide

### CODE REVIEW

**Comment by corneelwille** [comment]
On documentatie.uitdatabank.be there are currently 2 separate pages for sorting:
* one for places & events: https://documentatie.uitdatabank.be/content/search_api_3/latest/searching/offers/sorting.html
* one for organizers: https://documentatie.uitdatabank.be/content/search_api_3/latest/searching/organizers/sorting.html

However, a lot of documentation on these pages is redundant. So I merged the information from these pages into one sorting documentation page. If a sorting parameter is only applicable on a specific entity (e.g. events & places) I clarified that with an alert.

For now, the sorting parameter `popularity` (which is only used internally) is not documented yet until we've found a way to document these internal parameters on stoplight.

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [reply projects/uitdatabank/docs/search-api/sorting.md:22]: (I'd also suggest to reword "data" in the other parts of the guide to "results" or "events and places" or "organizers". Maybe "results" if it's all three, and "events and places" or "organizers" if it's only a subset?)

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:44]:
```suggestion
When searching by `geo distance`, it is possible to sort the results by the distance from the given coordinates.
```
Small typo here.

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:31]:
```suggestion
Sorts the data by the date it is created in the API.
```

(Every other sentence in the guide ends with a period, so I assume it's missing here by accident and not intentional)

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:74]:
```suggestion
The events with the heighest score will appear on top.
```

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:50]: I noticed that in some places you use "will appear on top", and in others "will appear first". I'd suggest to always use the same wording in "repetitive" parts. I have to look up the source but I read before that technical writing may be a bit more repetitive and does not need to switch up the words as much as in other prose, because it reduces possible confusion in technical documentation. (Now it may seem to some readers as if there's an actual difference between "first" and "on top".)

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:52]: Double newlines have the potential to make diffs harder to read later. I'll add it to the list of possible checks for the linter to add later. In the meantime it's not super bad if we sometimes have a double (or triple) newline somewhere as Stoplight will not render them anyway, but try to keep an eye out for them if you can to keep the cleanup later limited :)

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (bertramakers) [inline projects/uitdatabank/docs/search-api/sorting.md:14]: Hmm are these escapes necessary? 🤔

**Review by bertramakers — COMMENTED** [review]
Very nice for a first guide, no big remarks and looks very clear already to me 👍 I have just added some suggestions to make the little details perfect. ✨ (And also some things that we may want to agree on some guidelines on for other guides)

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (corneelwille) [reply projects/uitdatabank/docs/search-api/sorting.md]: Yes, I think it is logic to omit the environment URL 👍. I removed the host from the request examples now so that it is now consistent with the UiTPAS docs.

File: projects/uitdatabank/docs/search-api/sorting.md
Comment (corneelwille) [reply projects/uitdatabank/docs/search-api/sorting.md]: Those sentences were copied from the kramdown file for documentatie.uitdatabank.be, not sure if those escapes are actually needed there

**Comment by bertramakers** [comment]
After consulting with @corneelwille we have decided to move the commits for the default filters to a separate branch. We'll use this `uitdatabank/search-api-guides` PR/branch to merge all other guides into whenever they are done. And then we can merge this branch to merge everything at once when all the guides are done.

**Comment by bertramakers** [comment]
I've removed the commits about the default filters, and have pushed those to a separate branch (`uitdatabank/search-api-guides-default-filters). I'll convert this PR to a draft so we don't accidentally merge it before the other guides are ready and merged into this branch.

**Comment by bertramakers** [comment]
@corneelwille I've merged the Search API OpenAPI file into this PR (reviewed by Luc). You can go over it [here](https://docs.publiq.be/docs/uitdatabank/branches/uitdatabank%2Fsearch-api-guides/87dec20235b44-search-api). If there's anything missing or incorrect you can list it in a Jira ticket and I'll fix it!

**Review by bertramakers — APPROVED** [review]
Already approving this PR since everything in it has been reviewed at one point or another, and it will need to have an approval to be merged once everything is ready, and then I no longer get reminders in Slack to review this 😄

**Comment by bertramakers** [comment]
Merging this so we avoid having a super big PR that stays open for a long period of time. There are still some guides that need to be added, and we need for example a proper introduction page, and the OpenAPI endpoints can still use some finetuning here and there, but we can do that in separate follow-up PRs. The documentation so far is in this PR good enough to merge, especially since we don't actively link to this new documentation yet so we can still add/edit it.

**Pull request merged**
